flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -103,5 +103,10 @@ protected class RowDataDeltaWriter extends BaseEqualityDeltaWriter {
     protected StructLike asStructLike(RowData data) {
       return wrapper.wrap(data);
     }
+
+    @Override
+    protected StructLike asStructLikeKey(RowData data) {
+      return wrapper.wrap(data);
```

Comment:
@kbendick should this throw an exception like in https://github.com/apache/iceberg/blob/910f2712c7b731181c4a72fb6b8c66addeb76dd9/flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java#L109 or does it even matter what we return here for Flink 1.12?

Reply:
I would keep it this way upon initial inspection.

It needs to be implemented for the API and it will possibly get called. Keeping it the same as `asStructLike` seems safest for backwards compatibility.